eventuate-local-java-cdc-connector-mysql-binlog/src/main/java/io/eventuate/local/mysql/binlog/MySqlBinaryLogClient.java

```diff
@@ -126,71 +140,91 @@ protected void leaderStart() {
     client.setBinlogPosition(bfo.getOffset());
 
     client.setEventDeserializer(getEventDeserializer());
-    client.registerEventListener(event -> {
-      switch (event.getHeader().getEventType()) {
-        case TABLE_MAP: {
-          TableMapEventData tableMapEvent = event.getData();
-
-          if (cdcMonitoringDao.isMonitoringTableChange(tableMapEvent.getDatabase(), tableMapEvent.getTable())) {
-            cdcMonitoringTableId = Optional.of(tableMapEvent.getTableId());
-            tableMapEventByTableId.put(tableMapEvent.getTableId(), tableMapEvent);
-            break;
-          }
+    client.registerEventListener(event -> listenBinlogEventsHandleErrors(event, binlogFileOffset));
 
-          cdcMonitoringTableId = cdcMonitoringTableId.filter(id -> !id.equals(tableMapEvent.getTableId()));
+    connectWithRetriesOnFail();
 
-          SchemaAndTable schemaAndTable = new SchemaAndTable(tableMapEvent.getDatabase(), tableMapEvent.getTable());
+    try {
+      stopCountDownLatch.await();
+    } catch (InterruptedException e) {
+      handleProcessingFailException(e);
+    }
+  }
 
-          boolean shouldHandleTable = binlogEntryHandlers
-                  .stream()
-                  .map(BinlogEntryHandler::getSchemaAndTable)
-                  .anyMatch(schemaAndTable::equals);
+  private void listenBinlogEventsHandleErrors(Event event, Optional<BinlogFileOffset> binlogFileOffset) {
```

Comment:
This method should return immediately if `publishingException.isPresent() == true`. i think it's possible that the binlog client might call listenBinlogEventsHandleErrors() again.

Reply:
I will fix, but as I know client just stops listen events when exception is thrown